modules/account-lib/src/coin/xrp/walletInitializationBuilder.ts

```diff
@@ -1,26 +1,99 @@
 import { BaseCoin as CoinConfig } from '@bitgo/statics';
-import { BaseKey } from '../baseCoin/iface';
-import { NotImplementedError } from '../baseCoin/errors';
+import BigNumber from 'bignumber.js';
+import { TransactionJSON } from 'ripple-lib';
+import * as rippleTypes from 'ripple-lib/dist/npm/transaction/types';
+import RippleBinaryCodec from 'ripple-binary-codec';
+import { TransactionType } from '../baseCoin';
+import { InvalidTransactionError } from '../baseCoin/errors';
 import { Transaction } from './transaction';
 import { TransactionBuilder } from './transactionBuilder';
+import { WalletInitializationSchema } from './txnSchema';
 
 export class WalletInitializationBuilder extends TransactionBuilder {
+  protected _domain?: string;
+  protected _setFlag?: number;
+  protected _messageKey?: string;
+
   constructor(_coinConfig: Readonly<CoinConfig>) {
     super(_coinConfig);
   }
 
+  domain(domain: string): this {
+    this._domain = domain;
+    return this;
+  }
+
+  setFlag(setFlag: number): this {
+    this.validateValue(new BigNumber(setFlag));
+    this._setFlag = setFlag;
+    return this;
+  }
+
+  messageKey(messageKey: string): this {
+    this._messageKey = messageKey;
+    return this;
+  }
+
+  protected buildXRPTxn(): TransactionJSON {
```

Comment:
why not `buildImplementation()`?

Reply:
this is a helper function used in buildImplementation, varies based on builder. Creates an object that buildImplementation uses to create the transaction